### PR TITLE
Turn off cache on POST requests

### PR DESCRIPTION
**Pull request by sequoiaat**
Turn off cache control for the Connection object due to issues in apps created on Titanium as POST requests were failing. Proposing to turn cache off for the Connection object.

### CODE REVIEW

**Comment by aaaronlopez** [comment]
@sequoiaat can you remove the JAR from this commit? I will generate a new one when we prep release

**Comment by sequoiaat** [comment]
@sequoiaat Thank you for the quick review. I have reverted the Jar check-ins. Also saw your email on this topic. Thank you!

**Comment by aaaronlopez** [comment]
Great! Otherwise, it looks good.

@sequoiaat sorry, one more thing. Do you mind opening this against the `Staging` Branch

**Comment by sequoiaat** [comment]
@aaaronlopez Sure. Will do that.

**Comment by sequoiaat** [comment]
@aaaronlopez Just changed the base branch to **Staging** instead of **master**

File: Branch-SDK/src/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java [hunk 1]
```diff
@@ -129,6 +129,7 @@ private BranchResponse doRestfulPost(String url, JSONObject payload, int retryNu
             connection.setReadTimeout(timeout);
             connection.setDoInput(true);
             connection.setDoOutput(true);
+            connection.setUseCaches(false);
```
[the remark this answers is not in the document]
Comment (Sarkar) [reply Branch-SDK/src/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java]: Correct, but it was causing a bug in not showing the share sheet here: https://github.com/BranchMetrics/titanium-branch-deep-linking/issues/104#issuecomment-429831648

[the remark this answers is not in the document]
Comment (apeterson-branch) [reply Branch-SDK/src/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java]: Do we know 
A) that this fix actually resolves the customer's issue, and
B) that (with the caching "sticky") that another call in application code to `setDefaultUseCaches()` won't cause it to reappear?

**Comment by csalmi-branch** [comment]
I'm closing this PR, since I'm not sure it actually has a purpose anymore.

**Pull request closed without merging**